metal/id_test.go

```diff
@@ -0,0 +1,42 @@
+// Licenses under the MPL 2.0
+//
+// Source: https://github.com/hashicorp/terraform-plugin-sdk/pull/508
+package metal
+
+import (
+	"regexp"
+	"strings"
+	"testing"
+)
+
+var all36 = regexp.MustCompile(`^[a-z0-9]+$`)
+
+func TestPrefixedUniqueId(t *testing.T) {
+	prefix := "test-"
+	iterations := 10000
+	ids := make(map[string]struct{})
+	var id string
+	for i := 0; i < iterations; i++ {
+		id = PrefixedUniqueId(prefix)
+
+		if _, ok := ids[id]; ok {
+			t.Fatalf("Got duplicated id! %s", id)
```

Comment:
It's possible but improbable for this to fail. I suppose we can keep the sanity check.